pkg/dockerutil/dockerutils.go

```diff
@@ -159,14 +161,19 @@ func GetDockerClient() (context.Context, *dockerClient.Client) {
 		util.Debug("GetDockerClient: Setting DOCKER_HOST to '%s'", DockerHost)
 		_ = os.Setenv("DOCKER_HOST", DockerHost)
 	}
-	ctx := context.Background()
-	client, err := dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithAPIVersionNegotiation())
-	if err != nil {
-		output.UserOut.Warnf("Could not get Docker client. Is Docker running? Error: %v", err)
-		// Use os.Exit instead of util.Failed() to avoid import cycle with util.
-		os.Exit(100)
+	if DockerClient == nil {
+		DockerCtx = context.Background()
+		DockerClient, err = dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithAPIVersionNegotiation())
+		if err != nil {
+			output.UserOut.Warnf("Could not get Docker client. Is Docker running? Error: %v", err)
+			// Use os.Exit instead of util.Failed() to avoid import cycle with util.
+			os.Exit(100)
+		}
 	}
-	return ctx, client
+
+	defer DockerClient.Close()
```

Comment:
This should probably be inside `if DockerClient == nil`, but I'll wait for the tests to go through.